v1/fs/index.ts

```diff
@@ -0,0 +1,38 @@
+import path from 'path'
+import fs from 'fs'
+import tar from 'tar-fs'
+import gunzip from 'gunzip-maybe'
+
+export class SiteFileSystem {
+  path: string
+  constructor (rootPath: string) {
+    this.path = path.join(rootPath, 'sites')
+  }
+
+  async clear (siteId: string): Promise<void> {
+    const sitePath = this.getPath(siteId)
+    fs.rmSync(sitePath, { recursive: true, force: true })
+  }
+
+  getPath (siteId: string): string {
+    return path.join(this.path, siteId)
+  }
+
+  /// Reads a .tar or .tar.gz from given `tarballPath` and extracts it to
+  /// the target directory. Deletes original tarball when done
+  async extract (tarballPath: string, siteId: string): Promise<void> {
```

Comment:
Would it be possible to get a tarball stream instead of a path from the requests and process that instead?

Reply:
A streaming approach will generally be faster and gives us the opportunity to untar data right as it comes in. This means less disk usage overall and a faster time between submitting the tarball to getting the success response.

Though maybe that could be a problem when combining it with the size limit thing and with capabilities down the line. 🤔 Might be good to start with the fastest option and reevaluate as we go.

Reply:
I fixed the size limit thing with the streaming approach :) BusBoy has a `stream.truncated` option we can reject a promise on

Reply:
Actually, hm I just realized this does a partial overwrite of the folder which is not super safe, @RangerMauve thoughts?

Reply:
I think we can figure this out poat MVP. Might just want to take a note down for it